types/module/module.go

```diff
@@ -560,6 +571,14 @@ func (m *Manager) BeginBlock(ctx sdk.Context, req abci.RequestBeginBlock) abci.R
 		module, ok := m.Modules[moduleName].(BeginBlockAppModule)
 		if ok {
 			module.BeginBlock(ctx, req)
+			if m.consensusParamsGetter != nil && moduleName == "upgrade" {
```

Comment:
not sure any better way to know itself is upgrade module without circular import

Reply:
seems can rm this unnecessary check